pkg/operator/ceph/pool/controller.go

```diff
@@ -422,23 +422,44 @@ func configureRBDStats(clusterContext *clusterd.Context, clusterInfo *cephclient
 	logger.Debug("configuring RBD per-image IO statistics collection")
```

Comment:
> Maybe that's already taken care of?
> https://github.com/rook/rook/blob/master/pkg/operator/ceph/pool/controller.go#L247 &
> https://github.com/rook/rook/blob/master/pkg/operator/ceph/pool/controller.go#L293

Those links just point to the calling of this `configureRBDStats()` method. I don't see where this method will remove the pool from the list. All pools already in the list will remain in the list. So we will need a parameter to this method during deletion that tells which pool to remove from the list.

Reply:
I think this issue should be resolved from Ceph side itself, as any user setting this config should have this functionality of avoiding duplicates and non-existing entries. I've raised a PR for the same https://github.com/ceph/ceph/pull/48398

Reply:
If we wait for a fix from Ceph, Rook will need to behave differently for versions of ceph with or without the fix. Rook supports v16.2.0 and v17.2.0. I'd propose we can fix Rook without waiting for the ceph change. 

To remove the deleted pool from the list, the caller could pass the deleted pool name as a parameter to `configureRBDStats()`. The caller in the create/update case would just pass an empty value. Then inside this method, if the deleted pool name is set it would be removed from the list.